src/client.rs

```diff
@@ -226,19 +225,27 @@ impl StreamConnector for std::net::TcpStream {
     }
 
     fn to_stream_addr(url: Url) -> Result<Self::StreamAddr> {
+        use std::str::FromStr;
+        let Some(host) = url.host_str() else {
+            return Err(crate::error::Error::UrlError(String::from("no host")))
```

Comment:
to me these errors you are returning are a little confusing, IIUC the reason the host or port would be missing would be that the URL had some scheme that the library doesn't support, I think the correct error would be one saying like "unsupported URL scheme XXX"

Reply:
To deal with this scheme issue my first idea would be to make some wrapper struct like `HttpUrl(url::Url)` and its constructor could validate that it only contained HTTP urls. Then we could provide methods to to access things like the `host` which could panic if it didn't exist (because we previously validated that it must) This would also help make the examples stay shorter.
If you don't like that, perhaps then we could just have a function exposed they can call that validates the scheme like `let (host, port) = http_get_host_and_port(&url)?` or just  `ensure_http_url(&url)?`